src/libre/parser.act

```diff
@@ -90,11 +90,6 @@
 	#include "../comp.h"
 	#include "../../class.h"
 
-	struct flags {
-		enum re_flags flags;
-		struct flags *parent;
-	};
-
```

Comment:
So if we're setting these globally, and don't keep a stack or a tree, what's happening for cases like this?
```
; echo abcd | pcregrep '((?i:ABC))d'
abcd
; echo abcd | pcregrep '((?i:ABC))D'
;
```

I think we still need the idea of scope there because the flags do get restored outside of the capturing group.

Answer: I see you're saving and restoring the flags in pcre's parser.sid. That makes sense. So we're using the parser's call stack as our mechanism for scope.

This behaviour seems to apply to non-capturing groups, too. But we no longer need an AST node to hold that, because these are now set for every node. Okay! Great.